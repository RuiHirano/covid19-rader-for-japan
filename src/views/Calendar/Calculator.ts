import moment, { Moment } from 'moment'
import { Item, PeriodType } from '../../types';

interface QueryItemsProps {
	PeriodItems: Item[]
	BeforePeriodItems: Item[]
}

export class Calculator {
	items: Item[]
	periodType: PeriodType
	period: Moment
	periodItems: Item[]
	beforePeriodItems: Item[]
	constructor(items: Item[], periodType: PeriodType, period: Moment) {
		this.items = items
		this.periodType = periodType
		this.period = period
		const { PeriodItems, BeforePeriodItems } = this.queryItems()
		this.periodItems = PeriodItems
		this.beforePeriodItems = BeforePeriodItems
	}

	changePeriod(period: Moment) {
		this.period = period
		const { PeriodItems, BeforePeriodItems } = this.queryItems()
		this.periodItems = PeriodItems
		this.beforePeriodItems = BeforePeriodItems
		console.log(this.periodItems)
	}

	// itemsからperiod内のアイテムを取得する
	queryItems(): QueryItemsProps {
		let periodItems: Item[] = []
		let beforePeriodItems: Item[] = []
		for (var i = 0; i < (this.items).length; i++) {
			const item = this.items[i]
			switch (this.periodType) {
				case PeriodType.DAY:
					if (moment(item.StartDate).isSame(moment(this.period), 'days')) {
						periodItems.push(item)
					} else if (moment(item.StartDate).isBefore(moment(this.period), 'days')) {
						beforePeriodItems.push(item)
					}
					break
				case PeriodType.MONTH:
					if (moment(item.StartDate).isSame(moment(this.period), 'month')) {
						periodItems.push(item)
					} else if (moment(item.StartDate).isBefore(moment(this.period), 'month')) {
						beforePeriodItems.push(item)
					}
					break
				case PeriodType.YEAR:
					if (moment(item.StartDate).isSame(moment(this.period), 'year')) {
						periodItems.push(item)
					} else if (moment(item.StartDate).isBefore(moment(this.period), 'year')) {
						beforePeriodItems.push(item)
					}
					break
				case PeriodType.ALL:
					periodItems.push(item)
					break
			}

		}
		return { PeriodItems: periodItems, BeforePeriodItems: beforePeriodItems }
	}

	getItems(): Item[] {
		return this.items
	}

	getPeriodItems(): Item[] {
		return this.periodItems
	}

	getPeriodType(): PeriodType {
		return this.periodType
	}

	getPeriod(): Moment {
		return moment(this.period)
	}

	getStatistics() {

	}
}
