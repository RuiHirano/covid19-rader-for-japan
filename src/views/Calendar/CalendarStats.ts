import { MarketType, TradeType, Image, Item, PeriodType } from "../../types"
import uuid from "uuid"
import { Moment } from "moment"
import { ItemClass } from "../../types/item"
import { ItemsController } from "../../types/itemsController"

export class CalendarStats extends ItemsController {
	InitialInvethisment: number
	ProfitBefore: number
	TotalAssets: number
	NumDepWith: number
	AmountDepWith: number
	TotalProfitAndLoss: number
	TotalProfit: number
	TotalLoss: number
	WinRate: number
	NumTrade: number
	NumWin: number
	NumLose: number
	ProfitRatio: number
	ProfitAve: number
	LossAve: number

	constructor(items: ItemClass[], periodType: PeriodType, period: Moment, initialInvethisment: number) {
		super(items, periodType, period)
		this.InitialInvethisment = initialInvethisment
		this.ProfitBefore = 0
		this.TotalAssets = 0
		this.NumDepWith = 0
		this.AmountDepWith = 0
		this.TotalProfitAndLoss = 0
		this.TotalProfit = 0
		this.TotalLoss = 0
		this.WinRate = 0
		this.NumTrade = 0
		this.NumWin = 0
		this.NumLose = 0
		this.ProfitRatio = 0
		this.ProfitAve = 0
		this.LossAve = 0

		this.calcStatistics()
	}

	calcStatistics() {
		this.PeriodItems.forEach((item) => {
			if (item.TradeType === TradeType.DEPOSIT || item.TradeType === TradeType.WITHDRAWAL) {
				this.NumDepWith += 1
				this.AmountDepWith += item.Profit
			} else if (item.TradeType === TradeType.BUY || item.TradeType === TradeType.SELL) {
				if (item.Profit < 0) {
					this.TotalLoss += item.Profit
					this.NumLose += 1
				} else if (item.Profit >= 0) {
					this.TotalProfit += item.Profit
					this.NumWin += 1
				}
			}
		})

		this.BeforePeriodItems.forEach((item) => {
			if (item.TradeType === TradeType.BUY || item.TradeType === TradeType.SELL) {
				this.ProfitBefore += item.Profit
			}
		})

		// num trade
		this.NumTrade = this.NumWin + this.NumLose

		// total assets
		this.TotalAssets =
			this.TotalProfit +
			this.TotalLoss +
			this.InitialInvethisment +
			this.ProfitBefore

		// total
		this.TotalProfitAndLoss = this.TotalProfit + this.TotalLoss

		// win rate
		if (this.NumWin + this.NumLose !== 0) {
			this.WinRate =
				+((this.NumWin * 100) / (this.NumWin + this.NumLose))
					.toFixed(1)
		}

		// profit ave, lose ave
		if (this.NumWin !== 0) {
			this.ProfitAve = this.TotalProfit / this.NumWin
		}
		if (this.NumLose !== 0) {
			this.LossAve = this.TotalLoss / this.NumLose
		}

		// profit ratio
		if (this.LossAve === 0) {
			this.ProfitRatio = Infinity
		} else {
			this.ProfitRatio = -((this.ProfitAve * 100) / this.LossAve).toFixed(
				1
			)
		}
	}

}