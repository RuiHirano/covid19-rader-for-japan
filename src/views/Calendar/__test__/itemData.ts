import uuid from 'uuid/v1';
import { MarketType, TradeType, Item, Statistics } from '../../../types';
import moment from 'moment';

export const items = <Item[]>[
	{
		ID: uuid(),
		MarketType: MarketType.FX,
		StartDate: moment([2019, 10, 10]).toISOString(),
		EndDate: moment([2019, 10, 10]).toISOString(),
		TradeType: TradeType.BUY,
		Pair: "USD/JPY",
		Lot: 2.0,
		EntryRate: 1.0,
		LossCutRate: 1.0,
		SettleRate: 1.0,
		Profit: 1.0,
		BeforeComment: "good",
		AfterComment: "good",
		Tags: [],
		Images: [],
		UpdatedAt: moment([2019, 10, 10]).toISOString(),
		CreatedAt: moment([2019, 10, 10]).toISOString()
	},
	{
		ID: uuid(),
		MarketType: MarketType.FX,
		StartDate: moment([2019, 10, 10]).toISOString(),
		EndDate: moment([2019, 10, 10]).toISOString(),
		TradeType: TradeType.BUY,
		Pair: "USD/JPY",
		Lot: 2.0,
		EntryRate: 1.0,
		LossCutRate: 1.0,
		SettleRate: 1.0,
		Profit: 1.0,
		BeforeComment: "good",
		AfterComment: "good",
		Tags: [],
		Images: [],
		UpdatedAt: moment([2019, 10, 10]).toISOString(),
		CreatedAt: moment([2019, 10, 10]).toISOString()
	},
	{
		ID: uuid(),
		MarketType: MarketType.FX,
		StartDate: moment([2019, 11, 10]).toISOString(),
		EndDate: moment([2019, 11, 10]).toISOString(),
		TradeType: TradeType.BUY,
		Pair: "USD/JPY",
		Lot: 2.0,
		EntryRate: 1.0,
		LossCutRate: 1.0,
		SettleRate: 1.0,
		Profit: 1.0,
		BeforeComment: "good",
		AfterComment: "good",
		Tags: [],
		Images: [],
		UpdatedAt: moment([2019, 11, 10]).toISOString(),
		CreatedAt: moment([2019, 11, 10]).toISOString()
	},
	{
		ID: uuid(),
		MarketType: MarketType.FX,
		StartDate: moment([2019, 4, 10]).toISOString(),
		EndDate: moment([2019, 4, 10]).toISOString(),
		TradeType: TradeType.BUY,
		Pair: "USD/JPY",
		Lot: 2.0,
		EntryRate: 1.0,
		LossCutRate: 1.0,
		SettleRate: 1.0,
		Profit: 1.0,
		BeforeComment: "good",
		AfterComment: "good",
		Tags: [],
		Images: [],
		UpdatedAt: moment([2019, 4, 10]).toISOString(),
		CreatedAt: moment([2019, 4, 10]).toISOString()
	}
];

export const novemberItems = <Item[]>[
	{
		ID: uuid(),
		MarketType: MarketType.FX,
		StartDate: moment([2019, 10, 10]).toISOString(),
		EndDate: moment([2019, 10, 10]).toISOString(),
		TradeType: TradeType.BUY,
		Pair: "USD/JPY",
		Lot: 2.0,
		EntryRate: 1.0,
		LossCutRate: 1.0,
		SettleRate: 1.0,
		Profit: 1.0,
		BeforeComment: "good",
		AfterComment: "good",
		Tags: [],
		Images: [],
		UpdatedAt: moment([2019, 10, 10]).toISOString(),
		CreatedAt: moment([2019, 10, 10]).toISOString()
	},
	{
		ID: uuid(),
		MarketType: MarketType.FX,
		StartDate: moment([2019, 10, 10]).toISOString(),
		EndDate: moment([2019, 10, 10]).toISOString(),
		TradeType: TradeType.BUY,
		Pair: "USD/JPY",
		Lot: 2.0,
		EntryRate: 1.0,
		LossCutRate: 1.0,
		SettleRate: 1.0,
		Profit: 1.0,
		BeforeComment: "good",
		AfterComment: "good",
		Tags: [],
		Images: [],
		UpdatedAt: moment([2019, 10, 10]).toISOString(),
		CreatedAt: moment([2019, 10, 10]).toISOString()
	},
];

export const novemberStatistics = <Statistics>{
	InitialInvestment: 0,
	ProfitBefore: 1,
	TotalAssets: 3,
	NumDepWith: 0,
	AmountDepWith: 0,
	TotalProfitAndLoss: 2,
	TotalProfit: 2,
	TotalLoss: 0,
	WinRate: 100,
	NumTrade: 2,
	NumWin: 2,
	NumLose: 0,
	ProfitRatio: Infinity,
	ProfitAve: 1,
	LossAve: 0,
}
