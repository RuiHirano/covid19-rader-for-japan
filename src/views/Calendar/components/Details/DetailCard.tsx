import React, { useState, useEffect } from "react";
import { makeStyles, Theme } from "@material-ui/core/styles";
import { Grid, Card, CardContent, Typography, Button } from "@material-ui/core";
import { Item } from "../../../../types";

const useStyles = makeStyles((theme: Theme) => ({}));

interface Props {
    item: Item;
    toEdit: (item: Item) => void;
    toDetail: (item: Item) => void;
}

const DetailCard: React.FC<Props> = props => {
    const { item, toEdit, toDetail } = props;
    const classes = useStyles();

    return (
        <Card>
            <CardContent>
                <Button onClick={() => toEdit(item)}>Edit</Button>
                <Button onClick={() => toDetail(item)}>Detail</Button>
                <Typography variant="h6">{item.TradeType}</Typography>
                <Typography variant="h6">{item.StartDate}</Typography>
                <Typography variant="h6">{item.Pair}</Typography>
                <Typography variant="h6">{item.Lot}</Typography>
                <Typography variant="h6">{item.Profit}</Typography>
                <Typography variant="h6">{item.BeforeComment}</Typography>
            </CardContent>
        </Card>
    );
};

export default DetailCard;
