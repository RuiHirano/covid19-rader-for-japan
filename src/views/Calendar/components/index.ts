export { Details } from './Details';
export { Statistics } from './Statistics';