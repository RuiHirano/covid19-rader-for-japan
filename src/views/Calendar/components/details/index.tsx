import React, { useState, useEffect } from "react";
import { makeStyles, Theme } from "@material-ui/core/styles";
import { Grid, Card, CardContent, Typography } from "@material-ui/core";
import * as H from "history";
import PerfectScrollbar from "react-perfect-scrollbar";
import DetailCard from "./detail-card";
import moment, { Moment } from "moment";
import { Item } from "../../../../types";
import { withRouter, match } from "react-router";
import { styled } from "@material-ui/core/styles";

const useStyles = makeStyles((theme: Theme) => ({
    root: {
        backgroundColor: theme.palette.common.white,
        height: "100%"
    },
    tile: {
        backgroundColor: theme.palette.common.white,
        height: 125
    },
    detail: {
        height: "80%",
        display: "flex",
        backgroundColor: theme.palette.common.white
    },
    statistics: {
        backgroundColor: theme.palette.common.black,
        height: "20%"
    },
    calendar: {
        backgroundColor: theme.palette.common.white,
        height: "100%",
        width: "100%"
    }
}));

const Date = styled(Typography)({
    textAlign: "center"
});

interface Props {
    date: Moment;
    items: Item[];
    history: H.History;
    location: H.Location;
    match: match;
}

const Details: React.FC<Props> = props => {
    const { date, items, history } = props;
    const classes = useStyles();

    const toEdit = (item: Item) => {
        history.push("/entry/" + item.ID);
    };

    const toDetail = (item: Item) => {
        console.log("itemId: ", item);
        history.push("/detail/" + item.StartDate);
    };

    const queryDateItems = () => {
        let dateItems: Item[] = [];
        /*const iterator: Iterator = items.iterator()
		while(iterator.hasNext()){
			const item = iterator.next()
			if (moment(item.StartDate).isSame(moment(date), "days")) {
                dateItems.push(item);
            }
		}*/
        items.forEach(item => {
            if (moment(item.StartDate).isSame(moment(date), "days")) {
                dateItems.push(item);
            }
        });
        return dateItems;
    };

    const dateItems = queryDateItems();

    return (
        <div>
            <PerfectScrollbar>
                <Date variant="h6">{date.format("YYYY/MM/DD")}</Date>
                {dateItems.map((item, index) => (
                    <DetailCard
                        item={item}
                        toEdit={toEdit}
                        toDetail={toDetail}
                    />
                ))}
            </PerfectScrollbar>
        </div>
    );
};

export default withRouter(Details);
