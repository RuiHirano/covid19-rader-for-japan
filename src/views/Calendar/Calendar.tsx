import React, { useState, useEffect } from "react";
import { makeStyles, Theme } from "@material-ui/core/styles";
import Calendar from "react-calendar/dist/entry.nostyle";
import { width } from "@material-ui/system";
import { Grid, Typography } from "@material-ui/core";
import { Details, Statistics } from "./components";
import { Item, PeriodType } from "../../types";
import moment, { Moment } from "moment";
import { useSelector, useDispatch } from "react-redux";
import { AppState } from "../../redux/module/rootModule";
import { ItemClass } from "../../types/item";
import { CalendarStats } from "./CalendarStats";

// Container
interface ContainerProps {}
const CalendarContainer: React.FC<ContainerProps> = props => {
    const dispatch = useDispatch();
    const items = useSelector((state: AppState) => state.Items);
    console.log("items: ", items);

    /*const { isLoading, isFinishLoading } = useLoading(
        LoadingState.UPDATE_PASSWORD
    );
    useEffect(() => {
        if (isFinishLoading) {
            //history.push("/dashboard");
        }
    }, [isLoading]);*/

    return <CalendarView items={items} />;
};

export default CalendarContainer;

//Presentational
interface tileContentProps {
    date: Date;
    view: string;
}

interface Props {
    items: ItemClass[];
}

const CalendarView: React.FC<Props> = props => {
    const { items } = props;
    const classes = useStyles();

    const [date, setDate] = useState(new Date());

    const [stats, setStats] = useState(
        new CalendarStats(items, PeriodType.MONTH, moment(date), 10)
    );

    const getTileContent = ({ date, view }: tileContentProps) => {
        // 月表示のときのみ
        let content: any = [];
        items.forEach((item: ItemClass, index: number) => {
            if (moment(item.StartDate).isSame(moment(date), "days")) {
                content.push(
                    <Typography variant={"h6"}>{item.Pair}</Typography>
                );
            }
        });

        return <div>{content}</div>;
    };

    return (
        <div className={classes.root}>
            <Grid className={classes.root} container>
                <Grid item xs={12} sm={9}>
                    <Calendar
                        className={classes.calendar}
                        tileClassName={classes.tile}
                        value={date}
                        onClickDay={value => {
                            setDate(value);
                        }}
                        onActiveDateChange={values => {
                            console.log(values);
                            stats.changePeriod(moment(values.activeStartDate));
                            stats.calcStatistics();
                            setDate(values.activeStartDate);
                        }}
                        tileContent={getTileContent}
                        view={"month"}
                    />
                </Grid>
                <Grid item xs={12} sm={3}>
                    <Grid item className={classes.statistics} xs={12} sm={12}>
                        <Statistics date={moment(date)} stats={stats} />
                    </Grid>
                    <Grid item className={classes.detail} xs={12} sm={12}>
                        <Details date={moment(date)} items={items} />
                    </Grid>
                </Grid>
            </Grid>
        </div>
    );
};

const useStyles = makeStyles((theme: Theme) => ({
    root: {
        backgroundColor: theme.palette.common.white,
        height: "100%"
    },
    tile: {
        backgroundColor: theme.palette.common.white,
        height: 125
    },
    detail: {
        height: "100%",
        display: "flex",
        backgroundColor: theme.palette.common.white
    },
    statistics: {
        backgroundColor: theme.palette.common.white,
        height: "20%"
    },
    calendar: {
        backgroundColor: theme.palette.common.white,
        height: "100%",
        width: "100%"
    }
}));

//export default CalendarView;
