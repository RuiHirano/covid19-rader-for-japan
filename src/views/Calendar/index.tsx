import React, { useState, useEffect } from "react";
import { makeStyles, Theme } from "@material-ui/core/styles";
import Calendar from "react-calendar/dist/entry.nostyle";
import { width } from "@material-ui/system";
import { Grid, Typography } from "@material-ui/core";
import { Details, Statistics } from "./components";
import { Item, PeriodType, Items, StatsResult } from "../../types";
import moment, { Moment } from "moment";
import { useSelector, useDispatch } from "react-redux";
import { AppState } from "../../redux/module";
import { Iterator } from "../../types/iterator";

// Container
interface ContainerProps {}
const CalendarContainer: React.FC<ContainerProps> = props => {
    const dispatch = useDispatch();
    const items: Items = useSelector((state: AppState) => state.Items);

    const [date, setDate] = useState<Moment>(moment());

    const content = useSelector(
        (state: AppState) => state.User.Setting.Content
    );

    const [statsValues, setStatsValues] = useState(
        items.calculator(moment(), PeriodType.ALL, content)
    );

    useEffect(() => {
        setStatsValues(items.calculator(date, PeriodType.MONTH, content));
    }, [date]);

    const changeDate = (nextDate: Moment) => {
        setDate(nextDate);
    };

    return (
        <CalendarView
            items={items}
            statsValues={statsValues}
            date={date}
            changeDate={changeDate}
        />
    );
};

export default CalendarContainer;

//Presentational
interface tileContentProps {
    date: Date;
    view: string;
}

interface Props {
    items: Items;
    statsValues: StatsResult;
    date: Moment;
    changeDate: (date: Moment) => void;
}

const CalendarView: React.FC<Props> = props => {
    const { items, statsValues, date, changeDate } = props;
    const classes = useStyles();

    const getTileContent = ({ date, view }: tileContentProps) => {
        // 月表示のときのみ
        let content: any = [];
        /*const iterator: Iterator = items.iterator();
        while(iterator.hasNext()){
			const item = iterator.next()
			if (moment(item.StartDate).isSame(moment(date), "days")) {
                content.push(
                    <Typography variant={"h6"}>{item.Pair}</Typography>
                );
            }
		}*/
        items.items.forEach((item: Item, index: number) => {
            if (moment(item.StartDate).isSame(moment(date), "days")) {
                content.push(
                    <Typography variant={"h6"}>{item.Pair}</Typography>
                );
            }
        });

        return <div>{content}</div>;
    };

    return (
        <div className={classes.root}>
            <Grid className={classes.root} container>
                <Grid item xs={12} sm={9}>
                    <Calendar
                        className={classes.calendar}
                        tileClassName={classes.tile}
                        value={date.toDate()}
                        onClickDay={value => {
                            changeDate(moment(value));
                        }}
                        onActiveDateChange={values => {
                            console.log(values);
                            changeDate(moment(values.activeStartDate));
                        }}
                        tileContent={getTileContent}
                        view={"month"}
                    />
                </Grid>
                <Grid item xs={12} sm={3}>
                    <Grid item className={classes.statistics} xs={12} sm={12}>
                        <Statistics date={date} stats={statsValues} />
                    </Grid>
                    <Grid item className={classes.detail} xs={12} sm={12}>
                        <Details date={date} items={items} />
                    </Grid>
                </Grid>
            </Grid>
        </div>
    );
};

const useStyles = makeStyles((theme: Theme) => ({
    root: {
        backgroundColor: theme.palette.common.white,
        height: "100%"
    },
    tile: {
        backgroundColor: theme.palette.common.white,
        height: 125
    },
    detail: {
        height: "100%",
        display: "flex",
        backgroundColor: theme.palette.common.white
    },
    statistics: {
        backgroundColor: theme.palette.common.white,
        height: "20%"
    },
    calendar: {
        backgroundColor: theme.palette.common.white,
        height: "100%",
        width: "100%"
    }
}));

//export default CalendarView;
