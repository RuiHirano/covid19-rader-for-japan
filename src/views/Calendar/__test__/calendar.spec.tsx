import * as React from "react";
import renderer from "react-test-renderer";
import { configure, shallow, mount } from "enzyme";
import { Button, TextField, Input } from "@material-ui/core";
import CalendarView from "../Calendar";
import { Details } from "../components";
import DetailCard from "../components/Details/DetailCard";
import { Statistics } from "../../Report/components";
import { items } from "./itemData";
import Calendar from "react-calendar/dist/entry.nostyle";
import moment from "moment";
import { Item } from "../../../types";

describe("スクリーンショットテスト", () => {
    it("Snapshot of <DetailCard />", () => {
        const item: Item = items[0];
        const detailCard = shallow(
            <DetailCard item={item} toEdit={jest.fn()} toDetail={jest.fn()} />
        );
        expect(detailCard).toMatchSnapshot();
    });
    it("Snapshot of <Details />", () => {
        const details = shallow(<Details date={moment()} items={items} />);
        expect(details).toMatchSnapshot();
    });
    it("Snapshot of <Statistics />", () => {
        const statistics = shallow(<Statistics />);
        expect(statistics).toMatchSnapshot();
    });
});

describe("機能テスト", () => {
    //const calendarView = shallow(<CalendarView items={items} />);
    it("カレンダー内のセルを押すとトレードリストが変わる", () => {});
    it("カレンダーの日付が初期値では現在の日付となる", () => {
        //calendarView.instance().setDate(new Date());
    });
    it.todo(
        "モックデータがカレンダーへ反映される"
    ); /*, () => {
        const wrapper = shallow(<CalendarView items={items} />);
        console.log(wrapper.instance().getTileContent());

        expect("calenderText").toEqual("expectedValue");
    });*/
    it.todo("カレンダーの次へ、前へを押すと統計が変化する");
    it.todo("カレンダーの次へ、前へを押すとリストが変化する");
    it.todo("リストの編集ボタンを押すと編集画面へ遷移する");
    it.todo("リストのリストボタンを押すと詳細画面へ遷移する");
});
