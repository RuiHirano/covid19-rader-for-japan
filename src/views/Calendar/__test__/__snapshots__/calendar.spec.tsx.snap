// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スクリーンショットテスト Snapshot of <DetailCard /> 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      0
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      2019-11-09T15:00:00.000Z
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      USD/JPY
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      2.0
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      1.0
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      good
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`スクリーンショットテスト Snapshot of <Details /> 1`] = `
<div>
  <ScrollBar
    className=""
    component="div"
    containerRef={[Function]}
  >
    <WithStyles(ForwardRef(Typography))
      variant="h5"
    >
      Mon Nov 11 2019 11:30:02 GMT+0900
    </WithStyles(ForwardRef(Typography))>
  </ScrollBar>
</div>
`;

exports[`スクリーンショットテスト Snapshot of <Statistics /> 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={4}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={3}
    sm={6}
    xl={3}
    xs={12}
  >
    <TotalAssets />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={3}
    sm={6}
    xl={3}
    xs={12}
  >
    <TotalProfit />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={3}
    sm={6}
    xl={3}
    xs={12}
  >
    <WinRate />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={3}
    sm={6}
    xl={3}
    xs={12}
  >
    <ProfitRatio />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={8}
    md={12}
    xl={9}
    xs={12}
  >
    <ExpectedValue />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={4}
    md={6}
    xl={3}
    xs={12}
  >
    <BankruptRate />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
